### PR TITLE
A Pure MLP Architecture for Computer Vision

### LINKED ISSUE #2675: MLP-Mixer: A pure MLP architecture for computer vision
### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

MLP-Mixer: A pure MLP architecture for Vision

### Introduction paragraph (2-3 paragraphs):

Convolutional Neural Networks (CNNs) are the go-to model for computer vision. Recently, attention-based networks, such as the Vision Transformer, have also become popular. In this article, we show that while convolutions and attention are both sufficient for good performance, neither of them are necessary. The article discusses the MLP-Mixer, an architecture based exclusively on multi-layer perceptrons (MLPs).

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

By the end of this article, a reader should be able to understand:

1. An introduction and brief overview
2. The MLP-Mixer Architecture
3. The effects of Scale
4. The learned weights visualization

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

1. [Perceptron Algorithm - A Hands On Introduction](https://www.section.io/engineering-education/perceptron-algorithm/)
2. [MLP-Mixer: An all-MLP Architecture for Vision](https://arxiv.org/abs/2105.01601)

### PR DESCRIPTION
**Pull request by Bayler**
This fixes #2675

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
@Bayler 
Good afternoon and thank you for submitting your article for publishing. It has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

File: content/articles/mlp-mixer/index.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+#### The Mixer layers
+The mixer layer contains two types of MLP layers. It is important to note that the idea of these two layers is borrowed from CNN.
+
+1. Channel-mixing MLP
+
+- This MLP layer allows for the "mixing" of the per-location features.
+
+2. Token-mixing MLP
+
+- This MLP layer allows for "mixing" spatial information.
```
Comment (collins-a) [inline content/articles/mlp-mixer/index.md:94]: Which are these _per-location features_ and _spatial information_?

File: content/articles/mlp-mixer/index.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+#### The classification head
+This is the last layer in the MLP-Mixer architecture.
+
+It consists of a standard classification head, a global average pooling layer, and a linear classifier. 
+
+#### Extra
+1. Additionally, other standard components included in this architecture include dropout, skip-connections, and the Layer Norm applied on the channels.
+
+2. They used GeLU instead of the popular ReLU activation function. GELU's use is also evident in GPT-3, BERT, and most other newer Transformers.
```
Comment (collins-a) [inline content/articles/mlp-mixer/index.md:108]: Any particular reason as to why GeLU, as opposed to ReLU?

**Review by collins-a — COMMENTED** [review]
@Bayler well done. Near flawless article on MLP mixer. A couple of changes and we're going to the final review

**Comment by hectorkambow** [comment]
plagiarism check done

**Review by collins-a — APPROVED** [review]

**Comment by WanjaMIKE** [comment]
@ahmadmardeni1 ignore my review. I think it mistyped or something

**Comment by ahmadmardeni1** [comment]
Which review? @WanjaMIKE

**Comment by WanjaMIKE** [comment]
I have deleted it. Thanks

**Comment by ahmadmardeni1** [comment]
Good work  🎉 @Bayler @collins-a 

The URL should include keywords relevant for SEO. Please keep it in my mind for future articles.

**Pull request merged**